### DISCUSSION TITLE
ScenegraphLayer getShaders() not working

### DISCUSSION THREAD

**Discussion by euzu** [visgl/deck.gl#5206] [Q&A]
I have subclassed ScreengraphLayer to use another vertex shader. I implemented the getShaders() method.
With the latest release this doesnt works anymore. Now i need to overwrite _getModelOptions() and set  
<return value>.modelOptions.vs = myCustomShader.

Is this the desired behaviour ?

**Comment by Pessimistress** [comment]
Has that ever worked? AFAIK this layer never used `getShaders`

**Comment by euzu** [comment]
Yes it worked, i checked twice. i have a deployment with the previous version which uses the custom shader. I Updated he development environment to the latest version and the shader is ignore.

**Comment by Pessimistress** [comment]
What was the previous version you were using?

**Comment by euzu** [comment]
Regardless of the history, it should be possible to use custom shaders. The current implementation does not take getShaders() into account.
Should it stay that way ?

**Comment by Pessimistress** [comment]
Yes it should be fixed. I'm just trying to understand if this is a regression, as you implied in your original statement. If a regression happened in the current release, then we need to patch it in 8.3.x. If it never worked, then we'll likely address it in 8.4.0.

**Comment by euzu** [comment]
Sorry, I was mistaken.
I had overwritten the method getLoadOptions() and additionally getShaders().  Apparently getLoadOptions() was responsible for setting the custom shader.  getShaders() was ignored.

**Comment by Pessimistress** [comment]
This is fixed in 8.4.0-alpha and will be in 8.4.0
